src/_guides/language/language-tour.md

```diff
@@ -2271,33 +2247,43 @@ https://gist.github.com/0cb25997742ed5382e4a
 p?.y = 4;
 {% endprettify %}
 
-Some classes provide constant constructors. To create a compile-time
-constant using a constant constructor, use `const` instead of `new`:
 
-<?code-excerpt "misc/test/language_tour/classes_test.dart (const)"?>
+### Using constructors
+
+You can create an object using a *constructor*.
+Constructor names can be either <code><em>ClassName</em></code> or
+<code><em>ClassName</em>.<em>identifier</em></code>. For example,
+the following code creates `Point` objects using the
+`Point()` and `Point.fromJson()` constructors:
+
+<?code-excerpt "misc/test/language_tour/classes_test.dart (object-creation)" replace="/ as .*?;/;/g"?>
 {% prettify dart %}
-var p = const ImmutablePoint(2, 2);
+var p1 = Point(2, 2);
+var p2 = Point.fromJson({'x': 1, 'y': 2});
 {% endprettify %}
 
-<aside class="alert alert-info" markdown="1">
-**Dart 2 note:**
-You can sometimes omit the `const` before the constructor,
-but not in this example.
-For more information, read about _constant context_ in the
-[informal specification for implicit creation.](https://github.com/dart-lang/sdk/blob/master/docs/language/informal/implicit-creation.md)
-{% comment %}
-update-for-dart-2
+The following code has the same effect, but
+uses the optional `new` keyword before the constructor name:
 
-To omit the `const` before a constructor,
-you need to be in a `const` context,
-which requires at least one other use of `const`.
-[EXAMPLE OR LINK GOES HERE]
+<?code-excerpt "misc/test/language_tour/classes_test.dart (object-creation-new)" replace="/ as .*?;/;/g"?>
+{% prettify dart %}
+var p1 = new Point(2, 2);
+var p2 = new Point.fromJson({'x': 1, 'y': 2});
+{% endprettify %}
 
-[TODO: Once https://github.com/dart-lang/pub/issues/1807 is fixed,
-update code snippet and tests.]
-{% endcomment %}
+<aside class="alert alert-info" markdown="1">
+**Version note:** The `new` keyword became optional in Dart 2.
 </aside>
 
+Some classes provide constant constructors. To create a compile-time
+constant using a constant constructor, put the `const` keyword
+before the constructor name:
+
+<?code-excerpt "misc/test/language_tour/classes_test.dart (const)"?>
+{% prettify dart %}
+final p = const ImmutablePoint(2, 2);
+{% endprettify %}
+
 Constructing two identical compile-time constants results in a single,
```

Comment:
I know that this is from before, but maybe we should link to [identical()](#) here?

The definition is already given at the end of this file:

    [identical()]: {{site.dart_api}}/{{site.data.pkg-vers.SDK.channel}}/dart-core/identical.html

Reply:
I'm going to leave this out, because I'd rather not add more words.